modules/motions/Straight.h

```diff
@@ -0,0 +1,49 @@
+/**
+ * @file   Straight.h
+ * @brief  直進動作の親クラス
+ * @author HaruArima08
+ */
+
+#ifndef STRAIGHT_H
+#define STRAIGHT_H
+
+#include "Motion.h"
+#include "Pid.h"
+#include "Mileage.h"
+
+class Straight : public Motion {
+ public:
+  /**
+   * @brief コンストラクタ
+   * @param robot ロボット本体への参照
+   * @param _speed   目標速度[mm/s]
+   */
+  Straight(Robot& robot, double _speed);
+
+  /**
+   * @brief 直進する
+   */
+  void run();
+
+ protected:
+  /**
+   * @brief 直進する際の事前条件判定をする
+   */
+  virtual bool isMetPreCondition()
+  {
+    return true;  // 事前条件は常に満たす(子クラスで実装)
+  }
```

Comment:
これも純粋仮想関数にしたら動かないとかある？純粋仮想関数にした方が実装し忘れとかはないかな．
```suggestion
  virtual bool isMetPreCondition() = 0;
```